src/transformers/generation/flax_utils.py

```diff
@@ -580,6 +592,29 @@ def _greedy_search(
         # per batch-item holding current token in loop.
         sequences = jnp.full((batch_size, max_length), pad_token_id, dtype=jnp.int32)
         sequences = lax.dynamic_update_slice(sequences, input_ids, (0, 0))
+        if output_scores:
+            if hasattr(self.config, "vocab_size") and self.config.vocab_size is not None:
```

Comment:
Hey @gante, would love your second opinion here! Essentially, what we want to do is build a placeholder tensor for our **generation output scores** in Flax. This tensor is of dim `(batch size, seq len, vocab size)` - the first two dims we can get trivially from our inputs / generation config. Is it safe to assume we can do `model.config.vocab_size` to get the vocab size for our models? Or should we look at building the placeholder tensor _on the fly_ the first time we return a generation output in `greedy_search_body_fn`? Something like:
```python
logits = logits_processor(state.sequences, logits, state.cur_len)
next_tokens_scores = logits_processor(state.sequences, logits, state.cur_len)

if state.scores is None:
    vocab_size = logits.shape[-1]
    scores = jnp.ones((batch_size, max_length, vocab_size)) * np.array(-1.0e7)

next_token = jnp.argmax(next_tokens_scores, axis=-1)
tokens_scores = state.scores.at[:, state.cur_len, :].set(next_tokens_scores) if output_scores else None
...
```

Reply:
We actually run the first generate step out of the generation loop (see line 670 - can we handle this here perhaps?)

Reply:
If it is not a significant cause for performance degradation, I would go with the 2nd suggestion (building on the fly, after the first step). Sometimes `model.config.vocab_size` doesn't match the saved weights shape, for the reason written [here](https://twitter.com/karpathy/status/1621578354024677377)

Reply:
Awesome - do you have enough to go from here @hannan72? We can do it the second way where we get the logits shape from our first generate step which is done outside of the traced function:
https://github.com/huggingface/transformers/blob/9ade58f0555430cec851e307c83c3a56c4a77d0b/src/transformers/generation/flax_utils.py#L730-L732

Reply:
WDYT about this design decision @hannan72?